packages/caver-transaction/src/transactionHelper/transactionHelper.js

```diff
@@ -0,0 +1,167 @@
+    '0x08': TX_TYPE_STRING.TxTypeValueTransfer,
+    TxTypeFeeDelegatedValueTransfer: '0x09',
+    '0x09': TX_TYPE_STRING.TxTypeFeeDelegatedValueTransfer,
+    TxTypeFeeDelegatedValueTransferWithRatio: '0x0a',
+    '0x0a': TX_TYPE_STRING.TxTypeFeeDelegatedValueTransferWithRatio,
+
+    TxTypeValueTransferMemo: '0x10',
+    '0x10': TX_TYPE_STRING.TxTypeValueTransferMemo,
+    TxTypeFeeDelegatedValueTransferMemo: '0x11',
+    '0x11': TX_TYPE_STRING.TxTypeFeeDelegatedValueTransferMemo,
+    TxTypeFeeDelegatedValueTransferMemoWithRatio: '0x12',
+    '0x12': TX_TYPE_STRING.TxTypeFeeDelegatedValueTransferMemoWithRatio,
+
+    TxTypeAccountUpdate: '0x20',
+    '0x20': TX_TYPE_STRING.TxTypeAccountUpdate,
+    TxTypeFeeDelegatedAccountUpdate: '0x21',
+    '0x21': TX_TYPE_STRING.TxTypeFeeDelegatedAccountUpdate,
+    TxTypeFeeDelegatedAccountUpdateWithRatio: '0x22',
+    '0x22': TX_TYPE_STRING.TxTypeFeeDelegatedAccountUpdateWithRatio,
+
+    TxTypeSmartContractDeploy: '0x28',
+    '0x28': TX_TYPE_STRING.TxTypeSmartContractDeploy,
+    TxTypeFeeDelegatedSmartContractDeploy: '0x29',
+    '0x29': TX_TYPE_STRING.TxTypeFeeDelegatedSmartContractDeploy,
+    TxTypeFeeDelegatedSmartContractDeployWithRatio: '0x2a',
+    '0x2a': TX_TYPE_STRING.TxTypeFeeDelegatedSmartContractDeployWithRatio,
+
+    TxTypeSmartContractExecution: '0x30',
+    '0x30': TX_TYPE_STRING.TxTypeSmartContractExecution,
+    TxTypeFeeDelegatedSmartContractExecution: '0x31',
+    '0x31': TX_TYPE_STRING.TxTypeFeeDelegatedSmartContractExecution,
+    TxTypeFeeDelegatedSmartContractExecutionWithRatio: '0x32',
+    '0x32': TX_TYPE_STRING.TxTypeFeeDelegatedSmartContractExecutionWithRatio,
+
+    TxTypeCancel: '0x38',
+    '0x38': TX_TYPE_STRING.TxTypeCancel,
+    TxTypeFeeDelegatedCancel: '0x39',
+    '0x39': TX_TYPE_STRING.TxTypeFeeDelegatedCancel,
+    TxTypeFeeDelegatedCancelWithRatio: '0x3a',
+    '0x3a': TX_TYPE_STRING.TxTypeFeeDelegatedCancelWithRatio,
+}
+
+const CODE_FORMAT = {
+    EVM: '0x0',
+}
+
+/**
+ * Refines the array containing signatures.
+ * - Removes duplicate signatures.
+ * - Removes the default empty signature(['0x01', '0x', '0x']) included with other signatures.
+ * - For an empty signature array, return an array containing the default empty signature(['0x01', '0x', '0x']).
+ *
+ * @param {Array.<string>} rlpEncodedTxs - An array of RLP-encoded transaction strings.
+ * @return {string}
+ */
+const refineSignatures = (sigArray, isLegacy = false) => {
+    const set = new Set()
+    let result = []
+
+    const arrayOfSignatures = !_.isArray(sigArray[0]) ? [sigArray] : sigArray
+
+    for (const sig of arrayOfSignatures) {
+        if (sig.length > 0 && !utils.isEmptySig(sig)) {
+            const sigString = sig.join('')
+            if (!set.has(sigString)) {
+                set.add(sigString, true)
+                result.push(sig)
+            }
+        }
+    }
+
+    if (result.length === 0) result = [['0x01', '0x', '0x']]
+
+    return !isLegacy ? result : result[0]
```

Comment:
Hmm... What if we have two signatures at the end, which signature should be returned? anyone is okay?

Reply:
I think refineSignatures will called in signature setter, and signatures setter will called in appendSignatures.
For legacyTransaction, appendSignatures function will override, so i think the length of result will be 1.
But when user try to set signatures value in transaction manually, then this logic cannot cover. 
So i added to check the length of result, when legacyTransaction in here. :)